port/cpl_port.h

```diff
@@ -1161,6 +1161,14 @@ extern "C++"
     {
         return t;
     }
+
+    /** Emulates the C++20 .contains() method */
+    template <typename C, typename V>
+    inline bool contains(const C &container, const V &value)
+    {
+        return container.count(value) != 0;
```

Comment:
Hm, doesn't `count` always need to do too much? It cannot stop at the first match.

Reply:
not for std::set and std::map, which are up to now the only users of that

Reply:
> not for std::set and std::map

... unless some call triggers the second overload of count? (Taking an "equivalent" key instead of an exact one.)

> up to now the only users of that

This is a very generic function interface, but the implementation is specific. It gets attention (read: review) when added or modified. It is unlikely to get attention when a different use is added elsewhere. Including in software *using* GDAL.

Reply:
Thanks for pointing me to this new C++14 second overload / "equivalent" key concept. I have a bit of a hard time fully understanding the impact of that , but my understanding of https://stackoverflow.com/questions/20317413/what-are-transparent-comparators, is that it is only available if you define something like a std::key or std::map specially constructed with a std::less<>, so that shouldn't occur with the current state of the code base. That said in https://github.com/OSGeo/gdal/pull/10518 I'm proposing to use a std::find() based implementation to avoid any performance issue.

>  It is unlikely to get attention when a different use is added elsewhere. Including in software using GDAL.

This can't as this is protected by a ifdef GDAL_COMPILATION. So for GDAL internal use only